src/mlpack/core/data/map_policies/missing_policy.hpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file missing_policy.hpp
+ * @author Keon Kim
+ *
+ * Missing map policy for dataset info.
+ */
+#ifndef MLPACK_CORE_DATA_MAP_POLICIES_MISSING_POLICY_HPP
+#define MLPACK_CORE_DATA_MAP_POLICIES_MISSING_POLICY_HPP
+
+#include <mlpack/core.hpp>
+#include <unordered_map>
+#include <boost/bimap.hpp>
+#include <mlpack/core/data/map_policies/datatype.hpp>
+#include <limits>
+
+
+using namespace std;
+
+namespace mlpack {
+namespace data {
+
+/**
+ * Same as increment map policy, but does not change type of features.
+ */
+class MissingPolicy
+{
+ public:
+  // typedef of mapped_type
+  using mapped_type = double;
+
+  MissingPolicy()
+  {
+    // Nothing to initialize here.
+  }
+
+  explicit MissingPolicy(std::set<std::string> missingSet) :
+    missingSet(std::move(missingSet))
+  {
+    // Nothing to initialize here.
+  }
+
+
+  template <typename MapType>
+  mapped_type MapString(const std::string& string,
+                        const size_t dimension,
+                        MapType maps,
+                        std::vector<Datatype>& types)
+  {
+    // If this condition is true, either we have no mapping for the given string
+    // or we have no mappings for the given dimension at all.  In either case,
+    // we create a mapping.
+    const double NaN = std::numeric_limits<double>::quiet_NaN();
+    if (missingSet.count(string) != 0 &&
```

Comment:
I see what you mean.  My thinking was, everything in `missingSet` is something we might expect to be in `maps` by the end of loading.  So you can save a little bit of time by just putting everything in `missingSet` into `maps` pre-emptively.  The code as it is written here only adds things into `maps` from `missingSet` when it is seen, but I am not sure it is important to have a different, standalone `missingSet`.  What do you think?